src/fu-engine.c

```diff
@@ -4342,23 +4447,40 @@ fu_engine_usb_device_added_cb (GUsbContext *ctx,
 
 	/* can be specified using a quirk */
 	plugin_name = fu_device_get_plugin (device);
-	if (plugin_name != NULL) {
-		g_autoptr(GError) error = NULL;
-		FuPlugin *plugin = fu_plugin_list_find_by_name (self->plugin_list,
-								plugin_name, &error);
-		if (plugin == NULL) {
-			g_warning ("failed to find specified plugin %s: %s",
-				   plugin_name, error->message);
-			return;
-		}
-		if (!fu_plugin_runner_usb_device_added (plugin, device, &error)) {
-			g_warning ("failed to add USB device %04x:%04x: %s",
-				   g_usb_device_get_vid (usb_device),
-				   g_usb_device_get_pid (usb_device),
-				   error->message);
-		}
+	if (plugin_name == NULL)
+		return;
+	plugin = fu_plugin_list_find_by_name (self->plugin_list, plugin_name, &error_local);
+	if (plugin == NULL) {
+		g_warning ("failed to find specified plugin %s: %s",
+			   plugin_name, error_local->message);
+		return;
+	}
+
+	/* wait for hardware to settle */
+	if (self->loaded && fu_device_get_insert_delay (FU_DEVICE (device)) > 0) {
```

Comment:
Loaded isn't set until after enumerate was finished so the idea behind the new loaded check being that the device was plugged in before the engine started then it doesn't need an insert delay?

Might want to write this in a comment for futureselves to read.